octodns_scaleway/__init__.py

```diff
@@ -312,6 +346,30 @@ def _data_for_CNAME(self, _type, records):
 
         return record
 
+    def _data_for_DS(self, _type, records):
+        values = []
+        for record in records:
+            if "key_id" not in record:
+                # Invalid record
+                continue
+            if self.OLD_DS_FIELDS:
+                value = {
+                    'flags': record["key_id"],
+                    'protocol': record["algorithm"],
+                    'algorithm': ScalewayDSAlgorithm[record["digest"]["type"]].value,
+                    'public_key': record["digest"]["digest"],
+                }
+            else:
+                value = {
+                    'key_tag': record["key_id"],
+                    'algorithm': record["algorithm"],
+                    'digest_type': ScalewayDSAlgorithm[record["digest"]["type"]].value,
+                    'digest': record["digest"]["digest"],
+                }
+            values.append(value)
+        # The TTL is hardcoded and is not returned by the API
+        return {'type': _type, 'values': values, 'ttl': '3600'}
```

Comment:
We can't set the TTL for the DS record, and the API doesn't return the TTL. Based on my tests, I assume it is always set to 3600.